third_party/move/move-compiler-v2/tools/testdiff/src/main.rs

```diff
@@ -214,10 +203,14 @@ fn classify_tests(
     for_v2: bool,
 ) -> BTreeMap<TestKey, PathBuf> {
     let mut result = BTreeMap::new();
+    let split_match = Regex::new("^([^.]*)\\.([^.]*)$").unwrap();
```

Comment:
It would be preferable to use standard library methods like `extension` (https://doc.rust-lang.org/std/path/struct.Path.html#method.extension) and `file_prefix` here.

Reply:
I found the file extension methods to be not as reliable as something better-defined like a Regexp.  (Which I guess is actually Regex, so maybe not so well-defined.)

Reply:
In what way are they not as reliable? Those methods are well documented for cases like `blah.exp1.exp2` etc, whereas for a regex, I need to hunt a bit more to analyze the behavior.

Reply:
I first tried using those methods but seemed to see inconsistency with multiple-levels of extensions (`foo.exp1`.with_extension("exp2") versus `foo.exp`.set_extension("exp2")).  I may just have been confused, but when I just tried again I ran into other problems:

`extension` is only useable on a `Path`.  Once we grab the filename, it's an `osstr` and those methods don't work.   So I'm sticking with the current code, which is a little ugly, but at least it's isolated ugliness and does what I want.